src/components/MapBox.vue

```diff
@@ -218,7 +225,8 @@ export default {
           link.textContent = id;
           // Creates a click event for each button so that when clicked by the user, the visibility property
           // is changed as is the class (color) of the button
-          link.onclick = function(e) {
+            link.onclick = function(e) {
+              googleAnalytics('layers-menu', 'click', 'user clicked ' + id);
```

Comment:
Huh, so if theres already on click in the button functionality we can call GA like usual? Otherwise gotta do the runGoogleAnalytics method?

Reply:
Ya, it is another Vue scoping issue. Which makes me wonder if there is another way to set up our functions so that they all have access to the global scope. The Google Analytics plug in works really nicely for anything where we have direct access to the <a> tag or a button type deal, since it has a globally available object that we can use. Things get weird when we have to attach to a Vue component that doesn't have anything to add an @click to, like with the MapLayers icon. Once inside of the methods that created the buttons and things like that, I lost the ability to just use the GA plugin global variable (this.$ga).  So, not totally sure that the way I did things is the best way, but it work to bring the (this.$ga) into scope. So for now this is a start, maybe we will figure out something better down the line.